### ISSUE TITLE
Would this work for a Fargate AWS cluster?

### ISSUE THREAD

**Issue by filippo82** [jupyterhub/kubespawner#410]
### Proposed change
`kubespawner` for Fargate AWS cluster

### Alternative options
This exists https://github.com/uktrade/fargatespawner but it only mentions AWS ECS.

### Who would use this feature?
Anyone interested in running JupyterHub on a serverless cluster.

### (Optional): Suggest a solution
Honestly I have no clue :/ I've just started playing with JupyterHub on cloud but the Zero To Jupyterhub guide is outdated for AWS (EKS).

Thank you!

**Comment by filippo82** [comment]
Good day @yuvipanda and sorry for the slow reply. Yes, I did read the user guide of Fargate. It does indeed create a Kubernetes cluster. I am just getting started into the wonderful world of cloud stuff (currently learning AWS because of free credits) and still trying to figure out a few things. I'll go ahead and try how things go and then I'll report back here.

Cheers!

**Comment by DarkmatterVale** [comment]
I'm working on deploying JupyterHub on AWS EKS through a fargate-only cluster. I've run into some issues (documented [here](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1726)), not sure if they are related to the kubespawner or something else.

**Comment by filippo82** [comment]
Hi @DarkmatterVale thanks for your comment. I am still working on my workflow but I think I am getting there. Unfortunately, I am too busy these days to put it all together in a document. It will probably happen next week or the one after. I had a quick look at the issue you mentioned above ([this one](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1726)) and a very first guess is that, at a certain point, you will need to create a Fargate profile for the `jhub` namespace. Just a guess though...

**Comment by DarkmatterVale** [comment]
Hi @filippo82 thanks for the advice. I did create a fargate profile for the jhub namespace that is in the proper subnets. Have you been able to get jupyterhub running on Fargate in EKS?

**Comment by filippo82** [comment]
Hi @DarkmatterVale, sorry for the slow response but unfortunately I had no time to work on this issue over the past weeks :/ What about you?

Cheers.

**Comment by DarkmatterVale** [comment]
I also haven't been able to work on this for the past few weeks. I'm still hopeful this is possible to achieve...

**Comment by filippo82** [comment]
Hi @DarkmatterVale I still haven't been able to work more on this... In this other [issue](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1726), you write that:

> I am happily running a JupyterHub cluster on EKS using EC2 instances as the compute

Would you be able to share instructions/documentation to achieve this?
Thank you :)

**Comment by consideRatio** [comment]
My understanding is that Fargate is allowing for a subset of what k8s provides, but excludes for example DaemonSet's and such. So, in my mind, the answer to this question about KubeSpawner is that: Yes: kubespawner would work, assuming we can create Pod's and PVCs on Fargate and have them be honored.

I'm closing this as a support question that has got a good enough answer for now.

**Comment by filippo82** [comment]
👍

**Comment by Abhi94N** [comment]
I tried to deploy both with the fargate spawner config as well as kubespawner in a fargate only cluster but it did not work. I'm not a 100% sure on what the issue is but I was able to launch a single-user notebook from the hub using the k8s python client. 

My guess initially is that fargate does not allow for pods with elevated privileges such as NET_ADMIN because that was the issue I was getting when I tested this in gke autopilot but I disabled the blockiptables section in the helm chart as an attempt to remove the elevated privilege and tried that in fargate and my pod was still stuck in pending status. 

It wouldn't even create the fargate node.
